cmd/collectiondir/serve.go

```diff
@@ -465,14 +486,20 @@ func (cs *collectionServer) getDidsForCollection(c echo.Context) error {
 	collection := c.QueryParam("collection")
 	_, err := syntax.ParseNSID(collection)
 	if err != nil {
-		return c.String(http.StatusBadRequest, fmt.Sprintf("bad collection nsid, %s", err.Error()))
+		return c.JSON(http.StatusBadRequest, xrpcError{
+			Error:   "InvalidCollectionNSID",
+			Message: err.Error(),
```

Comment:
FWIW, I often forget to use `err.Error()` and do `fmt.Sprintf("%s", err)`; please do update/correct me if I forget.